### PR TITLE
Add a FV version of heat flux BC in the Navier Stokes module

### LINKED ISSUE #18433: Split heat flux boundary condition for porous flow energy equation
## Reason
The heat flux on the boundary of a porous media is split between the solid and fluid phase. The split may depend on porosity or thermal conductivities. A new FVBCs would handle this naturally

## Design
FVHeatFluxBC, specified for both phases

## Impact
Allow study of heated boundary cases with a known heat flux, to split between the two phases

@aprilnovak I would basically adapted the Pronghorn HeatFluxBC and copy pasted its documentation
I think this BC is general enough that it can be placed in the framework

Alternatively, there is a sister PR in Pronghorn that does the exact same thing but in Pronghorn.

### PR DESCRIPTION
**Pull request by GiudGiud**
Add related enum, test and documentation
Closes #18433

### CODE REVIEW

**Comment by GiudGiud** [comment]
@lindsayad for SANA

File: modules/navier_stokes/doc/content/source/fvbcs/NSFVHeatFluxBC.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# NSFVHeatFluxBC
+
+!syntax description /FVBCs/NSFVHeatFluxBC
+
+# Description
+
+Boundary condition splitting a constant heat flux among a two-phase fluid and solid domain
+based on a number of different models. There is no general consenus on the best boundary
+condition for all flows, and virtually all models for splitting a constant heat flux
+among multiple phases are quite crude \cite{alazmi}.
```
Comment (lindsayad) [inline modules/navier_stokes/doc/content/source/fvbcs/NSFVHeatFluxBC.md:10]:
```suggestion
among multiple phases are quite crude [!citep](alazmi).
```

File: modules/navier_stokes/include/fvbcs/NSFVHeatFluxBC.h [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  NSFVHeatFluxBC(const InputParameters & parameters);
+
+protected:
+  virtual ADReal computeQpResidual() override;
+
+  /// Value of the heat flux
+  const Real & _value;
+
+  /// Which phase this boundary condition is applied to, i.e. 'fluid' or 'solid'
+  const NS::phase::PhaseEnum _phase;
```
Comment (lindsayad) [inline modules/navier_stokes/include/fvbcs/NSFVHeatFluxBC.h:52]: Sigh. [General style guide](https://google.github.io/styleguide/cppguide.html#Namespace_Names) is to use lower case for namespaces which these new namespaces are doing. I wish that I had read the style guide earlier...but basically all of our namespaces to date have been capitalized. I don't love the mix of different capitalization but I'm also not going to say we shouldn't follow generally accepted styles for new namespaces. So in conclusion... I'm fine with this

File: modules/navier_stokes/src/fvbcs/NSFVHeatFluxBC.C [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  params.addRequiredParam<MooseEnum>("splitting", getSplittingEnum(), "type of splitting");
+
+  params.addRequiredParam<MooseEnum>(
+      "locality",
+      getLocalityEnum(),
+      "whether to use local (at the boundary) or global (domain-averaged) "
+      "parameter values");
+
+  params.addCoupledVar(NS::porosity, "porosity");
+
```
Comment (lindsayad) [inline modules/navier_stokes/src/fvbcs/NSFVHeatFluxBC.C:34]: I don't think we need all the new lines in here

File: modules/navier_stokes/src/fvbcs/NSFVHeatFluxBC.C [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+      "average_kappa", "postprocessor that provides domain-averaged fluid thermal dispersion");
+  params.addParam<PostprocessorName>(
+      "average_k_solid", "postprocessor that provides domain-averaged solid thermal conductivity");
+  params.addParam<PostprocessorName>(
+      "average_kappa_solid",
+      "postprocessor that provides domain-averaged solid effective thermal "
+      "conductivity");
+  params.addClassDescription("Constant heat flux boundary condition with phase splitting "
+                             "for fluid and solid energy equations");
+
```
Comment (lindsayad) [inline modules/navier_stokes/src/fvbcs/NSFVHeatFluxBC.C:49]:
```suggestion
```

File: modules/navier_stokes/src/fvbcs/NSFVHeatFluxBC.C [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+  return params;
+}
+
+NSFVHeatFluxBC::NSFVHeatFluxBC(const InputParameters & parameters)
+  : FVFluxBC(parameters),
+    _value(getParam<Real>("value")),
+    _phase(getParam<MooseEnum>("phase").getEnum<NS::phase::PhaseEnum>()),
+    _split_type(getParam<MooseEnum>("splitting").getEnum<NS::splitting::SplittingEnum>()),
+    _locality(getParam<MooseEnum>("locality").getEnum<NS::settings::LocalityEnum>()),
+
```
Comment (lindsayad) [inline modules/navier_stokes/src/fvbcs/NSFVHeatFluxBC.C:59]:
```suggestion
```

File: modules/navier_stokes/src/fvbcs/NSFVHeatFluxBC.C [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+                     : nullptr),
+    _average_kappa_s(_locality == NS::settings::global &&
+                             _split_type == NS::splitting::effective_thermal_conductivity
+                         ? &getPostprocessorValue("average_kappa_solid")
+                         : nullptr),
+    _average_kappa(_locality == NS::settings::global &&
+                           _split_type == NS::splitting::effective_thermal_conductivity
+                       ? &getPostprocessorValue("average_kappa")
+                       : nullptr),
+
```
Comment (lindsayad) [inline modules/navier_stokes/src/fvbcs/NSFVHeatFluxBC.C:80]:
```suggestion
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: modules/navier_stokes/src/fvbcs/NSFVHeatFluxBC.C (hunk 3 above)
Comment (GiudGiud) [inline modules/navier_stokes/src/fvbcs/NSFVHeatFluxBC.C:34]:
```suggestion
```

**Comment by lindsayad** [comment]
Looks like we don't have the bib entry for `alazmi`?

**Comment by GiudGiud** [comment]
I ll go fetch it from pgh

**Review by lindsayad — APPROVED** [review]

**Pull request merged**